client/lib/api/index.js

```
@@ -28,11 +28,15 @@ class API {
   }
 
   async login({ email, password }) {
+    let response = {};
     try {
-      const response = await this.req.post('/user/login', { email, password });
-      return response.data.token;
+      response = await this.req.post('/user/login', { email, password });
+      return response;
```

Is it possible to combine the two lines into one? Not familiar enough with ES6 but just curious

---

ya, should be able to pass it back in one go. Good catch.

---

why are we returning the entire response, instead of just the token?

---

if the response has issues it should raise the proper exception based on the status code, so that we clearly separate response management logic in the `client` and focus on how to display things in the `frontend`

---

ie as much as possible, return only the most relevant information, in this case the `token`

---

Do we not also want resp status codes? I actually modified it at the beginning since we are checking status codes at the client side for 200 before processing resp. I suppose I can make it just check for nulls of token value.

---

Yeah we don’t want to return it to the caller, since that means the caller needs to worry about status codes - abstract that away and throw informative exceptions

---

Don’t check for null token values, since if we don’t throw an exception, the value won’t be null

---

That way the flow on the front end could be:

Try -> catch -> print error

Instead of:

Try -> get status code -> 10 lines of error code logic